### PR TITLE
feat: support bitbucket triggers workflows by hooks

### PR DESCRIPTION
**Pull request by zhujian7**
BitBucket version smaller than 5.4 can not support webhooks, we
suppose these bitbucket can use hooks, like ScriptRunner, to send
a post request to cyclone server when some events happened, and
the cyclone server will handle the request and trigger workflows.

**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @hyy0322 @supereagle 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/handler/v1alpha1/webhook.go [hunk 1]
```diff
@@ -60,34 +66,91 @@ func HandleWebhook(ctx context.Context, tenant, eventType, integration string) (
...
-		if err != nil {
-			return newWebhookResponse(err.Error()), err
+		hookEvent := request.Header.Get(bitbucket.HookEventHeader)
+		tenants := make([]string, 0)
+		var terr error
+		if hookEvent == "true" {
+			serverAddress := strings.Trim(request.Header.Get(bitbucket.ServerAddressHeader), "/")
+			hookData := bitbucket.ParseHookEvent(request)
+
+			// Trigger workflows for all tenants
```
Comment (hyy0322) [inline pkg/server/handler/v1alpha1/webhook.go:76]: Why?

File: pkg/server/handler/v1alpha1/webhook.go [hunk 2]
```diff
@@ -45,6 +46,11 @@ func HandleWebhook(ctx context.Context, tenant, eventType, integration string) (
...
 	request := service.HTTPContextFrom(ctx).Request()
 
+	type EventDataWithIntegration struct {
+		*scm.EventData
+		Integration string
+	}
+	var allData = make(map[string][]*EventDataWithIntegration)
 	var data *scm.EventData
 
 	if request.Header.Get(github.EventTypeHeader) != "" {
```
Comment (hyy0322) [inline pkg/server/handler/v1alpha1/webhook.go:56]: Can we use switch case for this four types SCM

File: pkg/server/handler/v1alpha1/webhook.go (hunk 1 above)
Comment (hyy0322) [inline pkg/server/handler/v1alpha1/webhook.go:76]: Why?

File: pkg/server/handler/v1alpha1/webhook.go (hunk 2 above)
Comment (hyy0322) [inline pkg/server/handler/v1alpha1/webhook.go:56]: Can we use switch case for this four types SCM ?

Comment (zhujian7) [reply pkg/server/handler/v1alpha1/webhook.go:56]: No, the keys in the request header are different for every SCM.
- github.EventTypeHeader
- bitbucket.EventTypeHeader
- svn.EventTypeHeader

File: pkg/server/handler/v1alpha1/webhook.go (hunk 1 above)
Comment (zhujian7) [reply pkg/server/handler/v1alpha1/webhook.go:76]: The hook configured in the bitbucket server can not know which tenant will be triggered.

**Comment by zhujian7** [comment]
@hyy0322 PTAL

**Pull request merged**